### ISSUE TITLE
Sending traces to an IPv6 only host

### ISSUE THREAD

**Issue by bamorim** [open-telemetry/opentelemetry-erlang#595]
I have this setup where I'm hosting an app in Fly.io and I have a Grafana Agent deployed and it is only accessible through an IPv6-only domain. So I set `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT = "http://my-grafana-agent.internal:4317"`, however I'm getting a `nxdomain` error.

Is there a way to set `inet6` option on the exporter grpc client socket?

**Comment by tsloughter** [comment]
Sorry, not at the moment. I will fix it this weekend or next week. 

Are you able to use the `http_protobuf` exporter for the time being?

**Comment by bamorim** [comment]
@tsloughter yeah, for sure, don't worry. For that one, how can I setup to support ipv6?

**Comment by bamorim** [comment]
Also, slightly unrelated issued. I tried sending traces directly to Grafana Cloud and I couldn't set just using the env vars because it uses a basic auth for authentication, so my `OTEL_EXPORTER_OTLP_HEADERS` ended up like `OTEL_EXPORTER_OTLP_HEADERS="authorization=Basic abcdefa...bcdef="` due to the padding of the Base64 and that was causing the env var parser to fail because of this:

https://github.com/open-telemetry/opentelemetry-erlang/blob/dc77e6fe2fbbb39435bddb1bf06e4e9844c29737/apps/opentelemetry/src/otel_configuration.erl#L436C36-L442

Maybe it would be interesting to implement some sort of escaping? Or simply don't split with `all` and allow the value to have other `=`?

**Comment by tsloughter** [comment]
With http it should work by default with the latest releases. It checks ipv6 first then falls back to 4.

**Comment by tsloughter** [comment]
Ah, yea, needs a more complicated parser than just splitting. That'd be good to have.

**Comment by tsloughter** [comment]
Or no, not more complicated, just `leading` instead of `all`.

**Comment by cadebward** [comment]
Hello! Any movement on setting the `inet6` option?
